utils/RequestVerifier.ts

```diff
@@ -0,0 +1,59 @@
+import { SignatureInformation } from "./SignatureInformation";
+
+export class RequestVerifier {
+    public static readonly DEFAULT_TOLERANCE: number = 5 * 60 * 1000;
+
+    verifyRequestSignature(requestBody: string, requestHeader: string, signingSecret: string, tolerance: number = RequestVerifier.DEFAULT_TOLERANCE) {
+        this.checkRequestBody(requestBody);
+        this.checkRequestHeader(requestHeader);
+        this.checkSigningSecret(signingSecret);
+        this.checkTolerance(tolerance);
+        const info: SignatureInformation = new SignatureInformation(requestHeader);
+        this.verifyTolerance(info, tolerance);
+        this.verifySignature(info, requestBody, signingSecret);
+    }
+
+    checkRequestBody(requestBody: string) {
+        if (requestBody === "" || requestBody === null) {
+            throw new Error("Request Body cannot be empty or null");
+        }
+    }
+
+    checkRequestHeader(requestHeader: string) {
+        if (requestHeader === "" || requestHeader === null) {
+            throw new Error("Error with request header, Request header is empty");
+        }
+        else if (!requestHeader.includes("t=")) {
+            throw new Error("Error with request header, timestamp is not present");
+        }
+        else if (!requestHeader.includes("v1=")) {
+            throw new Error("Error with request header, signatures are not present");
+        }
+    }
+
+    checkSigningSecret(signingSecret: string) {
+        if (signingSecret === "" || signingSecret === null) {
+            throw new Error("Signing secret cannot be empty or null");
+        }
+    }
+
+    checkTolerance(tolerance: number) {
+        if (tolerance <= 0 || Number.isNaN(tolerance) || tolerance >= NaN) {
```

Comment:
what does `tolerance >= NaN` mean?

Reply:
I wanted to ensure tolerance would not be greater than a certain value, will have to replace NaN with what the max value of number is, will replace this in next commit

Reply:
`NaN` stands for Not A Number. Doing any kind of comparison other than `isNaN` will never resolve to what you need it to be. 
What you're looking for is `Number.MAX_SAFE_INTEGER` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER

Reply:
will change this in latest commit